### PR TITLE
Fix friendly embedding on amp-ad, and reenable test-amp-ad-fake.js

### PR DESCRIPTION
**Pull request by zombifier**
See #22572. Seems like it's now failing due to a regression (function being stripped from compiled code).

### CODE REVIEW

**Comment by jridgewell** [comment]
redirecting to @lannka

**Comment by lannka** [comment]
add @rsimha for reviewing as it might be related to the recent closure compiler change

File: extensions/amp-bind/0.1/bind-impl.js [hunk 1]
```diff
@@ -207,6 +207,7 @@ export class Bind {
   /**
    * @param {!Window} embedWin
    * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @nocollapse
```
Comment (rsimha) [inline extensions/amp-bind/0.1/bind-impl.js]: Thanks for figuring this out.

File: test/integration/test-amp-ad-fake.js [hunk 2]
```diff
@@ -35,36 +35,38 @@ describe('A4A', function() {
     },
     () => {
       // TODO(@zombifer, #21545): Unskip flaky test
-      it.skip('should layout amp-img, amp-pixel, amp-analytics', () => {
```
Comment (rsimha) [inline test/integration/test-amp-ad-fake.js]: Can you also unskip the other test in this file?

https://github.com/ampproject/amphtml/blob/63c2dfc4ab2dbde964fd347a9de6974535db1ee2/test/integration/test-amp-ad-fake.js#L95-L96

File: extensions/amp-bind/0.1/bind-impl.js (hunk 1 above)
Comment (rsimha) [reply extensions/amp-bind/0.1/bind-impl.js]: I just verified that this PR reinstates all 8 instances of `@nocollapse` that were inadvertently removed by #22446 (see https://github.com/ampproject/amphtml/pull/22446/commits/206027c763bfb227eea98d9b5ae630f7202777e5)

/cc @lannka

**Pull request merged**